app/jobs/google/search_progress_job.rb

```diff
@@ -9,10 +9,20 @@ def perform(user_id)
 
       raise ArgumentError unless user
 
-      SearchProgressChannel.broadcast_replace_to "toast_search_progress_stream:user_#{user.id}",
-                                                 target: 'toast_search_progress',
-                                                 partial: 'keywords/toast_search_progress',
-                                                 locals: { pending_count: user.keywords.pending.count }
+      SearchProgressChannel.broadcast_append_to "card_search_progress_stream:user_#{user.id}",
+                                                target: 'card_search_progress',
+                                                partial: 'keywords/card_search_progress',
+                                                locals: { presenter: presenter(user.keywords) }
+    end
+
+    private
+
+    def presenter(keywords)
+      Google::CardSearchProgressPresenter.new(
+        parsed_count: keywords.parsed.where('updated_at > ?', 12.hours.ago).count,
```

Comment:
@ankitkalia1195 @junan 
Is this too much logic inside a Job? I think it's a little bit overkill to have a dedicated query object for SearchProgress, and as it's not 100% frontend (the percentage is, but not the count), I did not want to put that query inside the presenter.

Any alternative is welcomed :) 
Thanks!

Reply:
@malparty Whats the logic behind last 12 hours? 🤔

Reply:
The idea was to display a progress bar (percentage that move from 1 value to 100%). So we need to compare how many keywords have been parsed to get a percent. If we compare `keywords.parsed.count`, the percentage will become quickly irrelevant due to parsing history (ex: 1000 keyword parsed, you upload a new file of 10x keywords, you expect 10% progress when a keyword is being parsed, not 0.001% :) )

I find it to be a good workaround to keep things simple, with
- stateless backend
- no need to store a value on the DB (whether it is a fileId or something else)

Any alternative is welcome of course :) Thanks @ankitkalia1195